sdk/test-utils/perf/src/managerProgram.ts

```diff
@@ -116,8 +116,15 @@ export class ManagerPerfProgram implements PerfProgram {
   }
 
   private createWorkers(): void {
-    const cpuOption = this.parsedOptions.cpus.value ?? 0;
+    let cpuOption = this.parsedOptions.cpus.value ?? 0;
     const parallels = this.parsedOptions.parallel.value ?? 1;
+
+    if (this.parsedOptions.profile.value) {
+      // Overriding to 1 core
+      // since there is no point in observing profiling artifacts of all the cores that do the same thing
+      cpuOption = 1;
+    }
+
```

Comment:
Can we display an error about incompatible args? It'd be nice if when someone writes `--cpus 64 --profile` we can explicitly call that out as broken, but infer `--cpus 1` as the default when profiling is enabled but _no_ CPUs argument was specified.